### PR TITLE
search: enable enterprise home panels by default

### LINKED ISSUE #14287: Enterprise home: enable feature flag
Enable feature flag by default on Sourcegraph Server (keep it disabled in Cloud)

### PR DESCRIPTION
**Pull request by limitedmage**
Fixes #14287

Requires https://github.com/sourcegraph/deploy-sourcegraph-dot-com/pull/3518 to be completed first

### CODE REVIEW

**Review by attfarhan — APPROVED** [review]

File: dev/global-settings.json [hunk 1]
```diff
@@ -1,8 +1,7 @@
 {
   // This is set from an environment variable
   "experimentalFeatures": {
-    "showRepogroupHomepage": true,
-    "showEnterpriseHomePanels": true
+    "showRepogroupHomepage": true
```
Comment (attfarhan) [inline dev/global-settings.json:4]: Should this be `showEnterpriseHomepage`?
```suggestion
    "showEnterpriseHomepage": true
```

**Review by attfarhan — CHANGES_REQUESTED** [review]

File: dev/global-settings.json (hunk 1 above)
Comment (limitedmage) [reply dev/global-settings.json:4]: This is the current name of the feature flag. I think there are arguments to be made for renaming it (eg. removing the `enterprise` word from it as we start to use it more in Cloud), but it's out of scope for this PR.

Comment (attfarhan) [reply dev/global-settings.json:4]: Oh sorry I mistyped `showEnterpriseHomePanels` to `showEnterpriseHomepage`. Doesn't this PR want to enable `showEnterpriseHomePanels ` by default? Why are we enabling `showRepogroupHomepage` by default in dev and not `showEnterpriseHomePanels`?

Comment (lguychard) [reply dev/global-settings.json:4]: This PR is switching the feature flag default for `showEnterpriseHomePanels` from `false` to `true`, so it doesn't need to be explicitly listed in `global-settings.json` anymore.

**Review by lguychard — APPROVED** [review]

**Comment by lguychard** [comment]
@limitedmage can you add a changelog entry? Thanks!

**Review by attfarhan — APPROVED** [review]

File: dev/global-settings.json (hunk 1 above)
Comment (attfarhan) [reply dev/global-settings.json]: Ah so sorry 🤦

**Pull request merged**